test/node/double_tests.js

```diff
@@ -35,4 +36,74 @@ describe('Double', function () {
       });
     }
   });
+
+  describe('specialValues', () => {
+    function twiceSerialized(value) {
+      let serializedDouble = BSON.serialize({ d: value });
+      let deserializedDouble = BSON.deserialize(serializedDouble, { promoteValues: true });
+      let newVal = deserializedDouble.d;
+      return newVal;
+    }
+
+    it('inf', () => {
```

Comment:
My general recommendation for the test cases with repetitive setup is to make an array of test case names and values, and then generate the test cases based on that. It'll also make it more obvious if something requires special handling.

e.g.,
```js
const testCases = [
  { name: 'NaN', value: NaN }, 
  { name: 'Infinity', value: Infinity }
];

for (const { name, value } of testCases) {
  it(`should <describe what it should do> when value is: ${name}`, () => {
    const outcome = someFunction(value);
    expect(outcome).to.equal(value);
  })
}
```

Reply:
Just added the table tests!